planner/core/exhaust_physical_plans.go

```diff
@@ -1470,11 +1474,26 @@ func (p *LogicalJoin) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]P
 		}
 	})
 
+	if prop.IsFlashOnlyProp() && ((p.preferJoinType&preferBCJoin) == 0 && p.preferJoinType > 0) {
```

Comment:
delete `prop.IsFlashOnlyProp() && `

Reply:
I don't fine the existing check before this position, can you help to point it out ?